app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :check_user_for_edit_post, only: %i[edit]
+  before_action :check_user_for_delete_post, only: %i[destroy]
+
+  def index
+    @posts = Post.all
+    @post = Post.new
+    @main_user = User.find(current_user.id)
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:success] = 'Post has been successfully created'
+      redirect_back(fallback_location: root_path)
+    else
+      render 'posts/index'
+    end
+  end
```

Comment:
Let's test for this commented out part of this method. 
Let's remember to always test for both negative and positive cases.

```ruby
  def create
    @post = current_user.posts.build(post_params)
    if @post.save
      flash[:success] = 'Post has been successfully created'
      redirect_back(fallback_location: root_path)
    # else
    #   render 'posts/index'
    end
  end
```

Reply:
I think it's not necessary to test the else part of this create action, I think i will delete it because there are a lot of checks to be done before a user reaches this level and if he succeeds in reaching this level, there is only one option, his post must be created. There is an authentication check before this action is access which is the only requirement to perform a post.

Reply:
for this reason, I am taking away the if statement completely.